### PR TITLE
Verify past prisoner details in Nomis

### PR DESCRIPTION
**Pull request by alan**
Reads in the Nomis API configuration and the number of days of most recent
prisoners to check and prints out json entry for each prisoner check saying if
it was found or had and error.

By reading the Nomis API configuration we should be able to test in production
before configuring the server.

### CODE REVIEW

File: lib/tasks/one_off.rake [hunk 1]
```diff
@@ -45,4 +45,86 @@ namespace :pvb do
...
+
+    nomis_host = cli.ask('Nomis host: ')
+    days_to_check = cli.ask('Number of days to check: ', Integer)
+
+    data = Prisoner.
+           where('created_at > ?', days_to_check.days.ago).
+           pluck(:number, :date_of_birth).
+           uniq
+
+    queue = Queue.new
```
Comment (StupidCodeFactory) [inline lib/tasks/one_off.rake]: 👍

File: lib/tasks/one_off.rake [hunk 2]
```diff
@@ -45,4 +45,86 @@ namespace :pvb do
...
+    data.each do |pair| queue << pair end
+    STDOUT.puts "Checking #{queue.size} prisoners"
+    STDOUT.print "\n"
+
+    pool = ConnectionPool.new(size: 5, timeout: 60) do
+      Nomis::Client.new(nomis_host, nomis_token, nomis_key)
+    end
+
+    workers = 1.upto(10).map { |i| new_worker(pool, queue, i) }
```
Comment (StupidCodeFactory) [inline lib/tasks/one_off.rake]: could have a bit more than 10 threads maybe if this is a bit long? (need to be careful not to DDOS Nomis though)

File: lib/tasks/one_off.rake [hunk 3]
```diff
@@ -45,4 +45,86 @@ namespace :pvb do
...
+    end
+
+    log[:timing] = RequestStore.store[:custom_log_items][:api]
+    log
+  end
+
+  def new_worker(pool, queue, thread_number)
+    Thread.new do
+      request_id = "check_prisoner_details_#{thread_number}"
+      RequestStore.store[:request_id] = request_id
```
Comment (StupidCodeFactory) [inline lib/tasks/one_off.rake]: not sure why this is useful?

Comment (alan) [inline lib/tasks/one_off.rake]: To find the requests easily in Kibana

Comment (StupidCodeFactory) [inline lib/tasks/one_off.rake]: how this being logger to kibana?

File: lib/tasks/one_off.rake (hunk 2 above)
Comment (alan) [inline lib/tasks/one_off.rake]: Yeah, let's see. I'll test in dev when it gets deployed, we'll also want to tweak the connection pool

File: lib/tasks/one_off.rake (hunk 3 above)
Comment (alan) [inline lib/tasks/one_off.rake]: The request_id gets sent as a header, and kibana parses it from the gateway logs

Comment (StupidCodeFactory) [inline lib/tasks/one_off.rake]: ah yes found it https://github.com/ministryofjustice/prison-visits-2/blob/master/app/services/nomis/client.rb#L48

**Pull request merged**